shuffle/cli/src/shared.rs

```diff
@@ -103,6 +107,86 @@ pub fn get_shuffle_dir() -> PathBuf {
     BaseDirs::new().unwrap().home_dir().join(".shuffle")
 }
 
+pub struct DevApiClient {
+    client: Client,
+}
+
+// Client that will make GET and POST requests based off of Dev API
+impl DevApiClient {
+    pub fn new(client: Client) -> Result<Self> {
```

Comment:
init with network: &Url, so that you don't need pass it into every func

Reply:
great idea!